cdap-app-fabric/src/test/java/io/cdap/cdap/internal/operation/OperationLifecycleManagerTest.java

```diff
@@ -0,0 +1,260 @@
+                  .setNamespace(testNamespace)
+                  .setFilter(
+                      new OperationRunFilter(OperationType.PULL_APPS, OperationRunStatus.FAILED))
+                  .build();
+          operationLifecycleManager.scanOperations(request, batchSize, d -> gotRuns.add(d));
+          expectedRuns =
+              testNamespaceRuns.stream()
+                  .filter(detail -> detail.getRun().getType().equals(OperationType.PULL_APPS))
+                  .filter(detail -> detail.getRun().getStatus().equals(OperationRunStatus.FAILED))
+                  .collect(Collectors.toList());
+          Assert.assertArrayEquals(expectedRuns.toArray(), gotRuns.toArray());
+        });
+  }
+
+  @Test
+  public void testGetOperation() throws Exception {
+    OperationRunDetail expectedDetail =
+        insertRun(testNamespace, OperationType.PUSH_APPS, OperationRunStatus.RUNNING);
+    String testId = expectedDetail.getRun().getId();
+    OperationRunId runId = new OperationRunId(testNamespace, testId);
+
+    TransactionRunners.run(
+        transactionRunner,
+        context -> {
+          OperationRunDetail gotDetail = operationLifecycleManager.getOperationRun(runId);
+          Assert.assertEquals(expectedDetail, gotDetail);
+          try {
+            operationLifecycleManager.getOperationRun(
+                new OperationRunId(Namespace.DEFAULT.getId(), testId));
+            Assert.fail("Found unexpected run in default namespace");
+          } catch (OperationRunNotFoundException e) {
+            // expected
+          }
+        },
+        Exception.class);
+  }
+
+  private static OperationRunDetail insertRun(
+      String namespace, OperationType type, OperationRunStatus status)
+      throws IOException, OperationRunAlreadyExistsException {
+    long startTime = runIdTime.incrementAndGet();
+    String id = RunIds.generate(startTime).getId();
+    OperationRun run =
+        OperationRun.builder()
+            .setRunId(id)
+            .setStatus(status)
+            .setType(type)
+            .setMetadata(
+                new OperationMeta(Collections.emptySet(), Instant.ofEpochMilli(startTime), null))
+            .build();
+    OperationRunId runId = new OperationRunId(namespace, id);
+    OperationRunDetail detail =
+        OperationRunDetail.builder()
+            .setSourceId(AppFabricTestHelper.createSourceId(sourceId.incrementAndGet()))
+            .setRunId(runId)
+            .setRun(run)
+            .setPullAppsRequest(input)
+            .build();
+    TransactionRunners.run(
+        transactionRunner,
+        context -> {
+          OperationRunStore operationRunStore = new OperationRunStore(context);
+          operationRunStore.createOperationRun(runId, detail);
+        },
+        IOException.class,
+        OperationRunAlreadyExistsException.class);
+    return detail;
+  }
+
+  private static List<OperationRunDetail> insertTestRuns() throws Exception {
```

Comment:
this is same in store also. we can separate this into a OperationTestBase class which in turn extends AppFabricTestBase

Reply:
Done. Thank you for the suggestion :)